subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/ForkingTestClassProcessor.java

```diff
@@ -110,7 +110,6 @@ RemoteTestClassProcessor forkProcess() {
         builder.applicationClasspath(classPath);
         builder.applicationModulePath(modulePath);
         options.copyTo(builder.getJavaCommand());
-        builder.getJavaCommand().getModularity().getInferModulePath().set(modulePath.iterator().hasNext());
```

Comment:
This was a bad solution to work around this specifically for JUnit5 test execution which is the only thing that currently makes use of "Module Workers" in practice.